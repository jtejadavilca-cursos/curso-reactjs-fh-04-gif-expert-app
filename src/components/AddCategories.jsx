import { useState } from 'react';

import PropTypes from 'prop-types';

export const AddCategories = ({fnAddCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => setInputValue( target.value );

    const onSubmit = (e) => {
        e.preventDefault();

        const newCategory = inputValue.trim();
        if(newCategory.length <=1) return;

        if( fnAddCategory(newCategory) ) setInputValue('');
    }
    return (
    <>
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type='text'
                placeholder='Buscar Gifs'
                value={inputValue}
                onChange={onInputChange} />
        </form>
    </>
    )
}


AddCategories.propTypes = {
    fnAddCategory: PropTypes.func.isRequired
}