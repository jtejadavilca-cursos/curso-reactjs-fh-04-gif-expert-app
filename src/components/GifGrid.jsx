import PropTypes from 'prop-types';

import { GifGridItem } from './index.components';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {

    const { images, isLoading } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>

            
            {
                isLoading && (<h2>Cargando...</h2>)
            }

            <div aria-label='card-grid' className='card-grid'>
                {
                    images.map(img => (
                        <GifGridItem title={img.title} url={img.url} key={img.id}/>
                    ))
                }
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}