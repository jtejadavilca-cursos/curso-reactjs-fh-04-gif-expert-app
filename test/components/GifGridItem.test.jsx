import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

const title = 'Saitama';
const url = 'http://localhost/q?Saitama';

describe('Pruebas de <GifGridItem />', () => {
    test('Debe hacer match con el snapshot', () => {

        const { container, getByText } = render(<GifGridItem title={title} url={url} />);

        expect( container ).toMatchSnapshot();

        expect( getByText(title) ).toBeTruthy();
    });


    test('Debe hacer match con el snapshot', () => {

        const { getByText, getByTestId } = render(<GifGridItem title={title} url={url} />);

        expect( getByText(title) ).toBeTruthy();

        // const { src, alt } = getByTestId('test-img');
        // expect( src ).toBe( url );
        // expect( alt ).toBe( title );

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });

    // test('Debe lanzar error al no mandar parÃ¡metros requeridos', () => {
    //     const { getByText, getByTestId } = render(<GifGridItem />);

    //     // expect( getByText(title) ).toBeTruthy();

    //     // expect( getByTestId('test-img').src ).toContain( url );
    // });
})