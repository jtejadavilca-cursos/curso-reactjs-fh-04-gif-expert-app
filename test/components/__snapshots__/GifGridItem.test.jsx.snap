// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      data-testid="test-img"
      src="http://localhost/q?Saitama"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
